airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -679,11 +697,11 @@ def __init__(
         self.check_interval = check_interval
         self.max_ingestion_time = max_ingestion_time
         self.check_if_job_exists = check_if_job_exists
-        if action_if_job_exists in ("increment", "fail"):
+        if action_if_job_exists in {"random", "increment", "fail"}:
```

Comment:
If I understand well, `increment` value is deprecated but if you set it, the behavior is the same as `random`? I am wondering if we should not keep the current behavior if `increment` is set

Reply:
yes your understanding is correct.
We can be extra cautious and keep the existing behavior, as it's a change, but it's not a "breaking" change.
I think the "increment" behavior is a bit dangerous because it's just problems waiting to happen if you use it, so I'd rather remove it entirely.

Reply:
Fine with that :)

Reply:
These will cause the jobs to have different names from now onward, ya? I think @ferruzzi modified this code somewhat recently and we tried to keep the naming the same for backwards compatibility (folks may have other logic that depends on the names being in the format they have been historically).
How to handle these low level breaking changes has been a hot topic lately in Airflow, I still don't think we've landed on a spot everyone agrees with. Thoughts?

Reply:
well there is the "every change is a breaking change" philosophy of course, and _maybe_ I can imagine a workflow where changing this would be breaking, but it'd be pretty far fetched.
I can always keep the existing behavior, but I'd put big deprecation warnings around it, because let's say you run 10 jobs a day, you'd be looking at throttling exceptions after 2-3 years, which is not such a long time.

Reply:
also, the "shape" of the naming is still the same, if a regex had been designed for the old code, the new code would pass it.

Reply:
This was my recent poke at this code from back in November:  https://github.com/apache/airflow/pull/27634

Reply:
I think that since we're adding in a deprecation warning that should be sufficient as users will have proper logs as to why they are getting unexpected naming.